feat(health): add wrapper for cordova-plugin-health

alx-andru opened the pull request:
Add support for [cordova-health-plugin](https://github.com/dariosalvi78/cordova-plugin-health)

ihadeed commented on src/plugins/health.ts:
```diff
@@ -0,0 +1,232 @@
+/**
+ * @param: startDate: {type: Date}, start date from which to get data
+ * @param: endDate: {type: Date}, end data to which to get the data
+ * @param: dataType: {type: String}, the data type to be queried (see below for supported data types)
```
Documentation for interfaces should look like this:

```
export interface QueryOptions { 
   ...
   /**
   * The data type to be queried
   */
   dataType: string;
   ...
}
```

Each property needs to have its definition right above it, there is no need to have the type there as it's defined in the typings and the doc generator is able to read that.

ihadeed commented on src/plugins/health.ts:
```diff
@@ -0,0 +1,232 @@
+  plugin: 'cordova-plugin-health',
+  pluginRef: 'navigator.health',
+  repo: 'https://github.com/dariosalvi78/cordova-plugin-health',
+  install: ''
```
`install: ''` should be removed

ihadeed commented on src/plugins/health.ts:
```diff
@@ -0,0 +1,232 @@
+   *
+   * @param {Array<String>} datatypes a list of data types you want to be granted access to
+   *
+   * @return successCallback: {type: function}, called if all OK
```
This should be:
```
@return {Promise<any>}
```

ihadeed commented on src/plugins/health.ts:
```diff
@@ -0,0 +1,232 @@
+   * @return successCallback: {type: function}, called if all OK
+   * errorCallback: {type: function(err)}, called if something went wrong,
+   * err contains a textual description of the problem
+
```
Extra lines like this one should be removed

ihadeed commented on src/plugins/health.ts:
```diff
@@ -1,60 +1,156 @@
- * @param: endDate: {type: Date}, end data to which to get the data
- * @param: dataType: {type: String}, the data type to be queried (see below for supported data types)
- */
+/*
```
This large block of documentation is not necessary. It won't be visible on the site anyways.

We can add a link in the main documentation block telling people to visit the plugin's repo to learn more about the valid data types. Or this table can be under `@advanced` tag. Example:

```
/**
* @name PluginName
* @description ...
* @usage ...
* @advanced
* Put the table here 
*/
```

ihadeed commented on src/plugins/health.ts:
```diff
@@ -201,7 +201,7 @@ export class Health {
    * This function is only available on Android.
    *
-   * @return successCallback: {type: function()}, called if the function was called
+   * @return  {Promise<any>} {type: function()}, called if the function was called
```
All of this should be removed: 
```
 {type: function()}, called if the function was called
     * errorCallback: {type: function(err)}, called if something went wrong		     * errorCallback: {type: function(err)}, called if something went wrong
```

All what it needs is `@return {Promise<any>}` 

You can add additional description such as: 
```
@returns {Promise<any>} returns a promise that resolves when (something happens) and rejects if (something else happens).
```

alx-andru commented:
@ihadeed thx for your review comments. I modified my fork and added the changes. I think I used the review process of github in a wrong way, please let me know if there are further changes or adjustments to make.
